### DISCUSSION TITLE
About file organization of BPF sources code in `skywalking-rover`

### DISCUSSION THREAD

**Discussion by tsint** [apache/skywalking#12756] [General]
Some similar codes are scattered in different places. For example, there are several go_tls.c files with only minor differences. Why is that? Extracting a common file might be a better way. What do you think about it?

**Comment by mrproliu** [comment]
You can extract some of the same code into the https://github.com/apache/skywalking-rover/tree/main/bpf directory. But for the probe code, you can extract some code to the `h` file.
